### PR TITLE
add blog authoring flow and slug support

### PR DESCRIPTION
**Pull request by ryanatkn**
Improves the blog using the code from this PR - https://github.com/ryanatkn/ryanatkn.com/pull/5

- designs the authoring flow - run `gro post` and then edit `src/routes/[slug]/N.svelte` where N is the index of the post
- support blog slug, deriving it from the post title, and it's mutable, so the canonical/immutable blog id continues to use the post index but we get the human readable version (if a user visits the index version, it redirects)
- [x] disable feed generation until we author the first blog post (make sure things work without a `routes/blog/[slug]/0.svelte`

followup:

- blog task args falling back to prompts for title, author, etc
- separate `$lib/blog` from `$lib/feed`
- support authors per entry
- add a task to bump the `date_modified` for a post by index (`gro bump 0`)
- generate JSON Feed as well as Atom (and fix/expand compatibility with the spec)
- extract to a reusable library

### CODE REVIEW

File: src/lib/blog/new.task.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import type {Task} from '@feltcoop/gro';
```
Comment (ryanatkn) [inline src/lib/blog/new.task.ts:1]: I originally had this as `src/blog.task.ts` so you'd run `gro blog` to create a new post, not sure if `new` or something else is better.

File: src/lib/blog/BlogPostFooter.svelte [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<script lang="ts">
+	import type {FeedItemData} from '$lib/blog/blogFeed';
+
+	export let post: FeedItemData;
+</script>
+
+<p>Want to discuss this post privately?</p>
```
Comment (greatbacon) [inline src/lib/blog/BlogPostFooter.svelte:7]: This phrasing feels weird. Maybe something more like?
```suggestion
<p>Want to let us know what you thought?</p>
```

File: src/routes/blog/+page.svelte [hunk 3]
```diff
@@ -1,18 +1,18 @@
 <script lang="ts">
-	import {feedData} from '$lib/blog/blogFeedData.js';
+	import {feed} from './feed';
```
Comment (greatbacon) [inline src/routes/blog/+page.svelte:2]: Are these feed imports supposed to be relative?

File: src/routes/toolmakersforum/+page.svelte [hunk 4]
```diff
@@ -23,7 +23,7 @@
 		<p>Toolmakers Forum is a podcast about tools & the people who make them.</p>
 		<p>
 			It's hosted by <a href="https://12mod12.com">Hamilton</a> &
```
Comment (greatbacon) [inline src/routes/toolmakersforum/+page.svelte:25]: Should we just link to our emails or mastadon accounts or not link to anything? Feels weird if mine is the only one pointing at a personal website.

**Review by greatbacon — APPROVED** [review]
Left a couple of comments re: some content details, but looks great!

File: src/routes/blog/+page.svelte (hunk 3 above)
Comment (ryanatkn) [reply src/routes/blog/+page.svelte:2]: yeah I've been hoping SvelteKit would support `$routes` so we can get absolute links, it's a limitation if you're importing something from `src/routes`

File: src/routes/toolmakersforum/+page.svelte (hunk 4 above)
Comment (ryanatkn) [reply src/routes/toolmakersforum/+page.svelte:25]: oops I screwed this up, that shouldn't link to felt.social below -- how about `Hamilton (Mastodon)` ? where Hamilton is your personal website link (this is how we cited Jane on the readme of felt-server)

Comment (ryanatkn) [reply src/routes/toolmakersforum/+page.svelte]: updated here: https://github.com/feltcoop/felt.social/pull/47/commits/7d087f6b10605da47459c22144830dbad8013a76

**Pull request merged**